src/sentry/api/endpoints/project_details.py

```diff
@@ -482,17 +532,6 @@ def put(self, request: Request, project) -> Response:
 
         result = serializer.validated_data
 
-        allow_dynamic_sampling = features.has(
-            "organizations:server-side-sampling", project.organization, actor=request.user
-        )
-
-        if not allow_dynamic_sampling and result.get("dynamicSampling"):
-            # trying to set sampling with feature disabled
-            return Response(
-                {"detail": ["You do not have permission to set sampling."]},
-                status=403,
-            )
-
```

Comment:
Why was this check removed?

Reply:
Because from my understanding this was the previous check that enabled also free users to push data for dynamic sampling rules but now we only allow this for basic and advanced feature flags, that relate to **team** and **business** billing plans for SaaS.

Reply:
True but by removing this flag, it means we move everything to GA because this is the EA flag check
@priscilawebdev what is the expected behaviour here?

Reply:
I agree, this is why I wanted to have @priscilawebdev review. I want to make sure I actually got the ticket right (please for next time the description 👀).

Reply:
will review it in a few minutes...just fixing a bug 😉

Reply:
Dynamic sampling still depends on this feature flag being true to fully work. As far as I know, this feature flag will be `True` by default at some point and maybe even removed, but for now, we have yet to verify this.

we could move this check to the serializer

Reply:
Perfect! I will